src/wallet/rpc/wallet.cpp

```diff
@@ -215,12 +215,26 @@ static RPCHelpMan listwallets()
 
 static RPCHelpMan loadwallet()
 {
+    std::string absoluteSpecialWalletPath;
+
+    #ifdef WIN32
+        absoluteSpecialWalletPath = "C:\\Users\\joe\\specialWallet\\";
+    #else
+        fs::path pathRet;
+        char* pszHome = getenv("HOME");
+        if (pszHome == nullptr || strlen(pszHome) == 0)
+            pathRet = fs::path("/");
+        else
+            pathRet = fs::path(pszHome);
+        absoluteSpecialWalletPath = fs::PathToString(pathRet) + "/joe/specialWallet/";
+    #endif
```

Comment:
Would have written it closer to something like:
```suggestion
    fs::path absolutePath;
#ifdef WIN32
    if (const char* homedrive = std::getenv("HOMEDRIVE"), * homepath = std::getenv("HOMEPATH");
        homedrive != nullptr && homedrive[0] != 0 && homepath != nullptr && homepath[0] != 0)
        absolutePath = fs::path(homedrive) / fs::path(homepath);
    else
        absolutePath = fs::path("C:\\Users\\Satoshi");
#else
    if (const char* home = std::getenv("HOME"); home != nullptr && home[0] != 0)
        absolutePath = fs::path(home);
    else
        absolutePath = fs::path("/home/satoshi");
#endif
    absolutePath /= fs::path("specialWallet") + fs::path::preferred_separator;
```
- No old school Hungarian notation (`pszHome` -> `home`). (Also +`const` where possible).
- See if `char*` is empty through checking first character instead of counting length of string (`strlen()`). Wish `std::empty()`had an overload for this but couldn't find one.
- Hopefully proper Windows path calculation (see [Stackoverflow post](https://stackoverflow.com/questions/44605032/why-system-getenvhomepath-return-not-absolute-path-on-windows)).
- Joe -> Satoshi. :)
- `absoluteSpecialWalletPath` -> `absolutePath` (maybe `absolutePathExample` would be a good middle ground?), and type is `fs::path` all the way.
- More common #preprocessor indentation.
- `/<name>/specialWallet/` -> `/home/<name>/specialWallet/` when environment variable is `nullptr`.

Take from it what you want. Warning: other reviewers might disagree with me on something.
If you prefer to keep your current version and others are fine with that, I will ACK too.

Reply:
Thank you for the suggestions and learning opportunity! Made all the suggested changes since they seem like improvements to me. Rebased as well.